### ISSUE TITLE
Make AudioFrame / VideoFrame more gc-friendly

### ISSUE THREAD

**Issue by jlaine** [mikeboers/PyAV#517]
@Mofef pointed out in https://github.com/mikeboers/PyAV/issues/317#issuecomment-467893423 that our `VideoFrame` contain reference cycles which prevent the garbage collector from reclaiming the frames as soon as they are unused. Let's try and improve this!

Here is a trivial example which currently has a fairly large memory (but stable) memory consumption:

```python
import av
  
with av.open('test.mp4') as container:
    for frame in container.decode(video=0):
        print(frame.pts)
```

On a 1080p video, RSS stabilizes around 480MB for me.

**Comment by jlaine** [comment]
Applying https://github.com/mikeboers/PyAV/issues/516, RSS drops to 85MB.

**Comment by jlaine** [comment]
Fixed in 75036f6772a2eb2b93a0f3bd7af3ff6b3138e3ce
